packages/flutter/lib/src/material/about.dart

```diff
@@ -472,6 +482,314 @@ class LicensePage extends StatefulWidget {
+        packageName,
+        bindings.map((int i) => data.licenses[i]).toList(growable: false),
+      ),
+    );
+  }
+
+  Widget _packagesList(
+    final BuildContext context,
+    final int selectedId,
+    final _LicenseData data,
+    final bool drawSelection,
+  ) {
+    return ListView(
+      children: <Widget>[
+        widget.about,
+        ...data.packages
+            .asMap()
+            .entries
+            .map<Widget>((MapEntry<int, String> entry) {
+          final String packageName = entry.value;
+          final int index = entry.key;
+          final List<int> bindings = data.packageLicenseBindings[packageName];
+          return _PackageListTile(
+            packageName: packageName,
+            index: index,
+            isSelected: drawSelection && entry.key == (selectedId ?? 0),
+            numberLicenses: bindings.length,
+            onTap: () {
+              widget.selectedId.value = index;
+              _MasterDetailFlow.of(context).openDetailPage(_DetailArguments(
+                packageName,
+                bindings.map((int i) => data.licenses[i]).toList(growable: false),
+              ));
+            },
+          );
+        }),
+      ],
+    );
+  }
+}
+
+class _PackageListTile extends StatelessWidget {
+  const _PackageListTile({
+    Key key,
+    this.packageName,
+    this.index,
+    this.isSelected,
+    this.numberLicenses,
+    this.onTap,
+}) : super(key:key);
+
+  final String packageName;
+  final int index;
+  final bool isSelected;
+  final int numberLicenses;
+  final GestureTapCallback onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    return Ink(
+      color: isSelected ? Theme.of(context).highlightColor : Theme.of(context).cardColor,
+      child: ListTile(
+        title: Text(packageName),
+        subtitle: Text(MaterialLocalizations.of(context).licensesPackageDetailText(numberLicenses)),
+        selected: isSelected,
+        onTap: onTap,
+      ),
+    );
+  }
+}
+
+class _LicenseData {
+  final List<LicenseEntry> licenses = <LicenseEntry>[];
+  final Map<String, List<int>> packageLicenseBindings = <String, List<int>>{};
+  final List<String> packages = <String>[];
+  // Special treatment for the first package since it should be the package
+  // for delivered application.
+  String firstPackage;
+
+  void addLicense(LicenseEntry entry) {
+    for (final String package in entry.packages) {
+      _addPackage(package);
+      packageLicenseBindings[package].add(licenses.length);
+    }
+    licenses.add(entry);
+  }
+
+  void _addPackage(String package) {
+    if (!packageLicenseBindings.containsKey(package)) {
+      packageLicenseBindings[package] = <int>[];
+      firstPackage ??= package;
+        packages.add(package);
+        packages.sort((String a, String b) {
+          // Make sure first package remains at front
+          if (a == firstPackage) {
+            return -1;
+          }
+          if (b == firstPackage) {
+            return -1;
```

Comment:
Doesn't one of these need to return `1` to keep the first package at the front? Seems like if it returns -1 for both a and b being the first package that it'll be ordered arbitrarily.

Reply:
Of course. If b is firstPackage, then a > b, therefore return 1. Thanks.